packages/passport-interface/src/api/makeRequest.ts

```diff
@@ -0,0 +1,119 @@
+import { getErrorMessage } from "@pcd/util";
+import urlJoin from "url-join";
+import { APIResult, GetResultValue, ResultMapper } from "./apiResult";
+import { POST } from "./constants";
+
```

Comment:
I like the shape of this pattern, with HTTP request and error-handling details centralized and separated from app logic.  The error-handling is the part I'm not sure I fully understand and think maybe could be cleaner.